client/klay_client.go

```diff
@@ -225,6 +225,9 @@ func (ec *Client) TransactionCount(ctx context.Context, blockHash common.Hash) (
 func (ec *Client) TransactionInBlock(ctx context.Context, blockHash common.Hash, index uint) (*types.Transaction, error) {
 	var json *rpcTransaction
 	err := ec.c.CallContext(ctx, &json, "klay_getTransactionByBlockHashAndIndex", blockHash, hexutil.Uint64(index))
+	if err != nil {
+		return nil, err
+	}
 	if err == nil {
```

Comment:
I think we don't need a nil check for `err` anymore after this update.

Reply:
Could you elaborate on what `I think we don't need a nil check for err anymore` means?

Reply:
I mean that since we check `err != nil` now, we don't need below `if err == nil`.

Reply:
Got it. Removed.